Sources/NIOCore/EventLoopFuture+AssumeIsolated.swift

```diff
@@ -125,6 +125,32 @@ public struct NIOIsolatedEventLoop {
         return .init(promise: promise, cancellationTask: { scheduled.cancel() })
     }
 
+    @discardableResult
+    @available(*, noasync)
+    @inlinable
+    public func scheduleCallback(
+        at deadline: NIODeadline,
+        handler: some NIOScheduledCallbackHandler
+    ) throws -> NIOScheduledCallback {
+        try self._wrapped._scheduleCallbackIsolatedUnsafeUnchecked(at: deadline, handler: handler)
+    }
+
+    @discardableResult
+    @available(*, noasync)
+    @inlinable
+    public func scheduleCallback(
+        in amount: TimeAmount,
+        handler: some NIOScheduledCallbackHandler
+    ) throws -> NIOScheduledCallback {
+        try self._wrapped._scheduleCallbackIsolatedUnsafeUnchecked(in: amount, handler: handler)
+    }
+
+    @inlinable
+    @available(*, noasync)
+    public func cancelScheduledCallback(_ scheduledCallback: NIOScheduledCallback) {
+        self._wrapped.cancelScheduledCallback(scheduledCallback)
```

Comment:
The checking strategy here is incoherent. Methods that are marked `unsafeUnchecked` should have asserts. Methods without that need `preconditions`.